### PR TITLE
Story/gtw 3325 support bank account destinations for pay to bank payments

### PR DESCRIPTION
**Pull request by ivan-lomakov**
### Jira link
-  https://checkout.atlassian.net/browse/GTW-3325

### Functional comments
- Extends "Payment -> Request a payment or payout -> destination" with additional parameter "bank_account". Actually all these fields are taken from "Instruments -> Create an instrument"

### CODE REVIEW

**Comment by ivan-lomakov** [comment]
@mathew-tuvesson-cko what do you think of this API spec? Is it ok for you?
bank_account

File: spec/components/schemas/Payments/RequestDestinations/02_PaymentBankAccountDestination.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+type: object
+description: An existing payment instrument
+required:
+  - country
```
Comment (roman-pavlov-cko) [inline spec/components/schemas/Payments/RequestDestinations/02_PaymentBankAccountDestination.yaml:4]: Should we require more fields than just `country` to fulfil bank account requirements in the Vault?

**Review by roman-pavlov-cko — COMMENTED** [review]
Noticed Get payment contract has changed. Let's double-check.

image

In general, I am not sure the validation rules are correctly reflected for the bank account instrument in both the Vault and the Payment section of the reference @mathew-tuvesson-cko

**Comment by ivan-lomakov** [comment]
> Noticed Get payment contract has changed. Let's double-check.
> 
> image
> 
> In general, I am not sure the validation rules are correctly reflected for the bank account instrument in both the Vault and the Payment section of the reference @mathew-tuvesson-cko

I have checked and there is the same behavior  in "crusoe" branch. There is only "card" option in "master" branch. I guess both options must be in the dropdown

File: spec/components/schemas/Payments/RequestDestinations/02_PaymentBankAccountDestination.yaml (hunk 1 above)
Comment (ivan-lomakov) [reply spec/components/schemas/Payments/RequestDestinations/02_PaymentBankAccountDestination.yaml:4]: I got such reply from Ramani Brown:

There are no global required fields for bank accounts - they change depending on the country and currency of the payment (e.g. one country might use a bank code and account number, but another may just need an iban)
We need those fields to validate the bank account details, which is performed by FxP, and they check the right fields for that country/currency have been provided. Worth checking what fields are required when you call FxP’s API. I suspect you can use the currency of the payment for example.

**Review by roberto-brancorhodes-cko — APPROVED** [review]
LGTM 👍

**Pull request merged**